### ISSUE TITLE
Problem with .merge method

### ISSUE THREAD

**Issue by bartgras** [ErwinM/acts_as_tenant#130]
I've noticed that after using acts_as_tenant sometimes SQL query is being cut.

Imagine simplified example with 2 models: `Car` - `has_many` - `Door`

Basically, this query is incorrect:
`Door.joins(:car).merge( Car.where(category_id: category_id) )`

But this one is correct:
`Door.joins(:car).where("cars.category_id = #{category_id}")`

Without acts_as_tenant both queries work fine.

Error I'm getting is (with SQL adapted to Car/Door example):
ActiveRecord::StatementInvalid:
       Mysql2::Error: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '' at line 1: 
```
SELECT FROM doors
INNER JOIN cars ON cars.id = doors.car_id
WHERE cars.account_id = 1 AND doors.account_id = 1 AND cars.category_id =
```

`account_id` comes from tenant gem

The `cars.category_id = ` is not a mistake. That's how the query is being cut.

I'm using rails (= 4.2.3)

**Comment by ErwinM** [comment]
That's odd. AaT does not create any custom sql. It instead relies fully on Rails' default_scope functionality. If you could provide me with an app that shows this behavior I could have a look...                                                                                                                                                                                                                                                                                                                                        Erwin                                                                                                                                                                                                                From: bartgrasSent: Friday, 29 January 2016 10:04To: ErwinM/acts_as_tenantReply To: ErwinM/acts_as_tenantSubject: [acts_as_tenant] Problem with .merge method (#130)I've noticed that after using acts_as_tenant sometimes SQL is being cut

Imagine simplified example with 2 models: Car - has_many - Door

Basically, this query is incorrect:
Doorjoins(:car)merge( Carwhere(category_id: category_id) )

But this one is correct:
Doorjoins(:car)where("carscategory_id = #{category_id}")

Without acts_as_tenant both queries work fine

Error I'm getting is (with SQL adapted to Car/Door example):
ActiveRecord::StatementInvalid:
       Mysql2::Error: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '' at line 1: 

SELECT FROM doors
INNER JOIN cars ON carsid = doorscar_id
WHERE carscategory_id =


This 
carscategory_id =
is not a mistake That's how the query is being cut

—Reply to this email directly or view it on GitHub.

**Comment by jebw** [comment]
Pretty sure this is the following bug affecting both Rails 4.2 and Rails 5.0

https://github.com/rails/rails/issues/20154

We've got our own multitenancy code which works a similar way and we're hitting this bug when rails merge's where clauses

**Comment by jebw** [comment]
This gist replicates the issue

https://gist.github.com/repinel/cc67944f2f47301b8aa5
